### PR TITLE
Teet 1205 file description

### PR DESCRIPTION
**Pull request by jkorpelin**
Could especially use some review for the file replacement stuff. Need to create the "full" filename based on the extension of the replacing file and description of the existing file to properly check.

### CODE REVIEW

File: app/common/src/cljc/teet/file/file_model.cljc [hunk 1]
```diff
@@ -26,23 +34,40 @@
...
+(defn valid-chars-in-description?
+  [description]
+  (every? allowed-chars description))
+
+(defn valid-description-length?
+  [description]
+  (>= max-description-length (count description)))
+
 (defn validate-file
   "Validate allowed file type and max size. Returns map with error description
```
Comment (erno) [inline app/common/src/cljc/teet/file/file_model.cljc:46]:
```suggestion
  "Validate allowed file type, legal chars + max length in description and max size. Returns map with error description
```

File: app/backend/src/clj/teet/file/file_commands.clj [hunk 2]
```diff
@@ -257,6 +261,18 @@
    :payload {id :db/id :as file}
    :project-id (project-db/file-project-id db id)
    :authorization {:document/overwrite-document {:db/id id}}
+   :pre [^{:error :description-too-long}
```
Comment (erno) [inline app/backend/src/clj/teet/file/file_commands.clj:264]: Could these use the validate-file fn?

File: app/frontend/src/cljs/teet/ui/file_upload.cljs [hunk 3]
```diff
@@ -112,52 +109,71 @@
              (not (< 0 sequence-number 10000)))
     {:error :invalid-sequence-number}))
 
+(defn common-file-validation
+  [file]
+  (println "file: " file)
```
Comment (erno) [inline app/frontend/src/cljs/teet/ui/file_upload.cljs:114]:
```suggestion
```

File: app/frontend/src/cljs/teet/file/file_controller.cljs [hunk 4]
```diff
@@ -118,73 +140,96 @@
...
                                                 :file/original-name]))
                      :previous-version-id (:db/id file)}
+           :error-event (partial ->UploadNewVersionError progress-atom)
            :result-event (fn [result]
                            (map->UploadFileUrlReceived
                              (merge result
                                     {:file-data (:file-object new-version)
                                      :on-success (common-controller/->Refresh)})))}))
 
+  UploadNewVersionError
```
Comment (erno) [inline app/frontend/src/cljs/teet/file/file_controller.cljs:150]: Was there other difference besides the refresh-fx call in this and FileUploadError? Could maybe same code be used for both if the refresh-fx is harmless here too? I don't know what it does so this is shooting in the dark :)

**Review by erno — APPROVED** [review]

File: app/frontend/src/cljs/teet/file/file_controller.cljs (hunk 4 above)
Comment (jkorpelin) [reply app/frontend/src/cljs/teet/file/file_controller.cljs:150]: The other one needs to control atom state of the form, to remove the in-progress state.

**Pull request merged**